diff --speed-large-files --minimal -ru Stage-3.2.2-Source.orig/libstage/CMakeLists.txt Stage-3.2.2-Source/libstage/CMakeLists.txt
--- Stage-3.2.2-Source.orig/libstage/CMakeLists.txt	2009-10-28 20:48:05.000000000 +0000
+++ Stage-3.2.2-Source/libstage/CMakeLists.txt	2011-05-24 16:57:05.000000000 +0100
@@ -60,18 +60,19 @@
              LINK_FLAGS "${FLTK_LDFLAGS}"	  
 )
 
-set( stagebinarySrcs main.cc )
+#set( stagebinarySrcs main.cc )
 
-add_executable( stagebinary ${stagebinarySrcs} )
+#add_executable( stagebinary ${stagebinarySrcs} )
 
 # When compiling stagebinarySrcs, pass CFLAGS from fltk-config
-set_source_files_properties( ${stagebinarySrcs} PROPERTIES COMPILE_FLAGS "${FLTK_CFLAGS}" )
+#set_source_files_properties( ${stagebinarySrcs} PROPERTIES COMPILE_FLAGS "${FLTK_CFLAGS}" )
 
-set_target_properties( stagebinary PROPERTIES OUTPUT_NAME stage )
+#set_target_properties( stagebinary PROPERTIES OUTPUT_NAME stage )
 
-target_link_libraries( stagebinary stage )
+#target_link_libraries( stagebinary stage )
 
-INSTALL(TARGETS stagebinary stage
+#INSTALL(TARGETS stagebinary stage
+INSTALL(TARGETS stage
 	RUNTIME DESTINATION bin
 	LIBRARY DESTINATION lib
 )
diff --speed-large-files --minimal -ru Stage-3.2.2-Source.orig/libstageplugin/CMakeLists.txt Stage-3.2.2-Source/libstageplugin/CMakeLists.txt
--- Stage-3.2.2-Source.orig/libstageplugin/CMakeLists.txt	2009-10-28 20:48:05.000000000 +0000
+++ Stage-3.2.2-Source/libstageplugin/CMakeLists.txt	2011-05-24 16:56:52.000000000 +0100
@@ -27,6 +27,7 @@
 		       stage
 		       ${PLAYER_LIBRARIES}
 		       ${OPENGL_LIBRARIES}
+		       fltk
 
 )
 
