diff --speed-large-files --minimal -ru gazebo-0.9.0.orig/CMakeLists.txt gazebo-0.9.0/CMakeLists.txt
--- gazebo-0.9.0.orig/CMakeLists.txt	2009-07-08 19:00:41.000000000 +0100
+++ gazebo-0.9.0/CMakeLists.txt	2011-04-06 09:31:16.685351786 +0100
@@ -11,6 +11,8 @@
 SET (GAZEBO_VERSION 0.9.0)
 MESSAGE (STATUS "${PROJECT_NAME} version ${GAZEBO_VERSION}")
 
+ADD_DEFINITIONS(-D__STDC_CONSTANT_MACROS)
+
 SET (ENABLE_SHADOWS OFF CACHE BOOL "If ON, must apply ogre_patch.diff to Ogre sources")
 SET (BUILD_GAZEBO ON CACHE INTERNAL "Build Gazebo" FORCE)
 
diff --speed-large-files --minimal -ru gazebo-0.9.0.orig/player/FiducialInterface.cc gazebo-0.9.0/player/FiducialInterface.cc
--- gazebo-0.9.0.orig/player/FiducialInterface.cc	2009-07-08 19:00:41.000000000 +0100
+++ gazebo-0.9.0/player/FiducialInterface.cc	2011-04-06 09:33:26.986469944 +0100
@@ -33,7 +33,7 @@
 
 #include <math.h>
 #include <iostream>
-#include <libplayerxdr/playerxdr.h>
+#include <libplayerinterface/playerxdr.h>
 #include <boost/thread/recursive_mutex.hpp>
 
 #include "gazebo.h"
diff --speed-large-files --minimal -ru gazebo-0.9.0.orig/player/IRInterface.cc gazebo-0.9.0/player/IRInterface.cc
--- gazebo-0.9.0.orig/player/IRInterface.cc	2009-07-08 19:00:41.000000000 +0100
+++ gazebo-0.9.0/player/IRInterface.cc	2011-04-06 09:33:26.986469944 +0100
@@ -30,7 +30,7 @@
 
 #include <math.h>
 #include <iostream>
-#include <libplayerxdr/playerxdr.h>
+#include <libplayerinterface/playerxdr.h>
 #include <boost/thread/recursive_mutex.hpp>
 
 #include "gazebo.h"
diff --speed-large-files --minimal -ru gazebo-0.9.0.orig/player/LaserInterface.cc gazebo-0.9.0/player/LaserInterface.cc
--- gazebo-0.9.0.orig/player/LaserInterface.cc	2009-07-08 19:00:41.000000000 +0100
+++ gazebo-0.9.0/player/LaserInterface.cc	2011-04-06 09:33:26.986469944 +0100
@@ -29,7 +29,7 @@
 
 #include <math.h>
 #include <iostream>
-#include <libplayerxdr/playerxdr.h>
+#include <libplayerinterface/playerxdr.h>
 #include <boost/thread/recursive_mutex.hpp>
 
 #include "gazebo.h"
