diff --speed-large-files --minimal -ru mercury-compiler-10.04.1.orig/boehm_gc/libatomic_ops/src/atomic_ops/sysdeps/gcc/x86_64.h mercury-compiler-10.04.1/boehm_gc/libatomic_ops/src/atomic_ops/sysdeps/gcc/x86_64.h
--- mercury-compiler-10.04.1.orig/boehm_gc/libatomic_ops/src/atomic_ops/sysdeps/gcc/x86_64.h	2010-02-23 06:28:41.000000000 +0000
+++ mercury-compiler-10.04.1/boehm_gc/libatomic_ops/src/atomic_ops/sysdeps/gcc/x86_64.h	2012-01-20 16:48:19.000000000 +0000
@@ -119,7 +119,7 @@
   /* Note: the "xchg" instruction does not need a "lock" prefix */
   __asm__ __volatile__("xchgb %0, %1"
                 : "=q"(oldval), "=m"(*addr)
-                : "0"(0xff), "m"(*addr) : "memory");
+                : "0"((unsigned char) 0xff), "m"(*addr) : "memory");
   return (AO_TS_VAL_t)oldval;
 }
 
