diff --speed-large-files --minimal -ru peekabot-0.8.6.orig/m4/boost.m4 peekabot-0.8.6/m4/boost.m4
--- peekabot-0.8.6.orig/m4/boost.m4	2011-05-29 18:59:31.000000000 +0100
+++ peekabot-0.8.6/m4/boost.m4	2012-11-02 11:49:46.968983261 +0000
@@ -403,7 +403,7 @@
       LDFLAGS=$boost_save_LDFLAGS
       LIBS=$boost_save_LIBS
       if test x"$Boost_lib" = xyes; then
-        Boost_lib_LDFLAGS="-L$boost_ldpath -Wl,-R$boost_ldpath"
+        Boost_lib_LDFLAGS="-L$boost_ldpath"
         Boost_lib_LDPATH="$boost_ldpath"
         break 6
       else
diff --speed-large-files --minimal -ru peekabot-0.8.6.orig/src/Main.cc peekabot-0.8.6/src/Main.cc
--- peekabot-0.8.6.orig/src/Main.cc	2011-01-27 22:56:31.000000000 +0000
+++ peekabot-0.8.6/src/Main.cc	2012-11-02 11:49:46.968983261 +0000
@@ -77,7 +77,7 @@
      * \brief The formatter used to format the log messages.
      */
     const PrintfFormatter m_formatter;
-    
+
 public:
     Console() throw() : m_is_active(false), m_formatter("%m\n") {}
 
@@ -147,7 +147,7 @@
     }
     catch(po::error &e)
     {
-        std::cerr << "Error parsing command line arguments: " 
+        std::cerr << "Error parsing command line arguments: "
                   << e.what() << std::endl;
         return -1;
     }
@@ -203,7 +203,10 @@
     fs::create_dir_if_inexistant(fs::get_resource_path());
     fs::create_dir_if_inexistant(fs::get_resource_path() / "data");
     fs::create_dir_if_inexistant(fs::get_resource_path() / "snapshots");
-    fs::create_dir_if_inexistant(fs::get_temp_path() / "peekabot");
+    // this is a way to guarantee a unique temp dir to avoid problems occuring when running with different users (marc@hanheide.net)
+    std::string uniqName="peekabot";
+    uniqName=uniqName+"."+getenv("USER");
+    fs::create_dir_if_inexistant(fs::get_temp_path() / uniqName );
 
     //
     // Load configuration and setup default options
@@ -215,7 +218,7 @@
 
 
     //
-    // Log various settings and data that might be useful in dismissing 
+    // Log various settings and data that might be useful in dismissing
     // bug reports from users ;)
     //
     TheMessageHub::instance().publish(
@@ -229,11 +232,11 @@
         PEEKABOT_PBAR_COMPATIBLE_VERSION_STRING + " and later");
 
     TheMessageHub::instance().publish(
-        LOG_MESSAGE, "Resource directory: " + 
+        LOG_MESSAGE, "Resource directory: " +
         fs::get_resource_path().string());
 
     TheMessageHub::instance().publish(
-        LOG_MESSAGE, "Pkgdatadir: " + 
+        LOG_MESSAGE, "Pkgdatadir: " +
         fs::get_pkgdata_path().string());
 
     TheMessageHub::instance().publish(
diff --speed-large-files --minimal -ru peekabot-0.8.6.orig/src/UploadCache.cc peekabot-0.8.6/src/UploadCache.cc
--- peekabot-0.8.6.orig/src/UploadCache.cc	2010-09-08 12:52:48.000000000 +0100
+++ peekabot-0.8.6/src/UploadCache.cc	2012-11-02 11:49:46.972985211 +0000
@@ -155,7 +155,10 @@
 
 boost::filesystem::path UploadCache::get_upload_directory()
 {
-    return fs::get_temp_path() / "peekabot" / "uploads";
+    std::string uniqName="peekabot";
+    uniqName=uniqName+"."+getenv("USER");
+
+    return fs::get_temp_path() / uniqName / "uploads";
 }
 
 
diff --speed-large-files --minimal -ru peekabot-0.8.6.orig/src/utils/obj2pbmf/obj2pbmf.cc peekabot-0.8.6/src/utils/obj2pbmf/obj2pbmf.cc
--- peekabot-0.8.6.orig/src/utils/obj2pbmf/obj2pbmf.cc	2010-09-08 12:52:48.000000000 +0100
+++ peekabot-0.8.6/src/utils/obj2pbmf/obj2pbmf.cc	2012-11-02 11:50:30.150020673 +0000
@@ -473,9 +473,9 @@
     {
         boost::progress_display pd( 3*m_tri_count );
 
-        for( int i = 0; i < 3*m_tri_count; i++ )
+        for( int iter = 0; iter < 3*m_tri_count; iter++ )
         {
-            IndexTuple &triple = m_src_indices[i];
+            IndexTuple &triple = m_src_indices[iter];
 
             int i = triple.get<0>();
             int j = triple.get<1>();
Only in peekabot-0.8.6/src/utils/obj2pbmf: obj2pbmf.cc~
