diff --speed-large-files --minimal -ru OpenNI-OpenNI-d50d0e3.orig/Platform/Linux-x86/CreateRedist/install.sh OpenNI-OpenNI-d50d0e3/Platform/Linux-x86/CreateRedist/install.sh
--- OpenNI-OpenNI-d50d0e3.orig/Platform/Linux-x86/CreateRedist/install.sh	2011-04-18 15:18:53.000000000 +0200
+++ OpenNI-OpenNI-d50d0e3/Platform/Linux-x86/CreateRedist/install.sh	2011-05-03 13:03:17.000000000 +0200
@@ -1,9 +1,10 @@
 #!/bin/sh -e
 
-INSTALL_LIB=/usr/lib
-INSTALL_BIN=/usr/bin
-INSTALL_INC=/usr/include/ni
-INSTALL_VAR=/var/lib/ni
+INSTALL_LIB=$2/lib
+INSTALL_BIN=$2/bin
+INSTALL_INC=$2/include/ni
+INSTALL_VAR=$2/var/lib/ni
+
 
 OS_NAME=`uname -s`
 
@@ -101,7 +102,7 @@
 	fi
 	;;
 *) 
-	echo "Usage: $0 [options]"
+	echo "Usage: $0 [options] [install-path]"
 	echo "Available options:"
 	printf "\t-i\tInstall (default)\n"
 	printf "\t-u\tUninstall\n"
