diff --speed-large-files --minimal -ru OpenNI-OpenNI-d07a88a.orig/Source/OpenNI/Linux-x86/XnUSBLinux-x86.cpp OpenNI-OpenNI-d07a88a/Source/OpenNI/Linux-x86/XnUSBLinux-x86.cpp
--- OpenNI-OpenNI-d07a88a.orig/Source/OpenNI/Linux-x86/XnUSBLinux-x86.cpp	2011-10-17 02:14:46.000000000 +0100
+++ OpenNI-OpenNI-d07a88a/Source/OpenNI/Linux-x86/XnUSBLinux-x86.cpp	2011-11-24 23:24:32.122529307 +0000
@@ -1054,7 +1054,7 @@
 								if (pTransfer->buffer + nTotalBytes != pBuffer)
 								{
 //									printf("buffer %d has %d bytes. Moving to offset %d...\n", i, pPacket->actual_length, nTotalBytes);
-									memcpy(pTransfer->buffer + nTotalBytes, pBuffer, pPacket->actual_length);
+									memmove(pTransfer->buffer + nTotalBytes, pBuffer, pPacket->actual_length);
 								}
 								nTotalBytes += pPacket->actual_length;
 							}
