diff --speed-large-files --minimal -ru Stage-3.2.2-Source.orig/libstageplugin/CMakeLists.txt Stage-3.2.2-Source/libstageplugin/CMakeLists.txt
--- Stage-3.2.2-Source.orig/libstageplugin/CMakeLists.txt	2009-10-28 20:48:05.000000000 +0000
+++ Stage-3.2.2-Source/libstageplugin/CMakeLists.txt	2011-05-24 16:56:52.000000000 +0100
@@ -27,6 +27,7 @@
 		       stage
 		       ${PLAYER_LIBRARIES}
 		       ${OPENGL_LIBRARIES}
+		       fltk
 
 )
 
