diff -urN golem-trunk/compiler/cmake/default/CMakeLists.txt golem-trunk-patched/compiler/cmake/default/CMakeLists.txt
--- golem-trunk/compiler/cmake/default/CMakeLists.txt	2011-04-09 10:37:53.000000000 +0200
+++ golem-trunk-patched/compiler/cmake/default/CMakeLists.txt	2012-06-26 15:41:56.885765388 +0200
@@ -243,7 +243,7 @@
 	${GOLEM_ROOT}/include/Golem/Ctrl/Waypoint.h
 )
 ADD_LIBRARY(GolemCtrl SHARED ${CTRL_SOURCES})
-TARGET_LINK_LIBRARIES(GolemCtrl GolemDefs GolemMath GolemSys GolemTools)
+TARGET_LINK_LIBRARIES(GolemCtrl GolemDefs GolemMath GolemSys GolemTools dl)
 INSTALL(TARGETS GolemCtrl LIBRARY DESTINATION lib)
 INSTALL(FILES ${CTRL_HEADERS} DESTINATION include/Golem/Ctrl/)
 endif(BUILD_CTRL)
diff -urN golem-trunk/include/Golem/Math/Collection.h golem-trunk-patched/include/Golem/Math/Collection.h
--- golem-trunk/include/Golem/Math/Collection.h	2010-11-22 02:24:58.000000000 +0100
+++ golem-trunk-patched/include/Golem/Math/Collection.h	2012-06-26 15:35:55.009756786 +0200
@@ -12,6 +12,7 @@
 
 //------------------------------------------------------------------------------
 
+#include <stdlib.h>
 #include <vector>
 #include <list>
 #include <map>
diff -urN golem-trunk/resources/Device/GolemDeviceKatana450.xml golem-trunk-patched/resources/Device/GolemDeviceKatana450.xml
--- golem-trunk/resources/Device/GolemDeviceKatana450.xml	2011-04-21 22:24:36.000000000 +0200
+++ golem-trunk-patched/resources/Device/GolemDeviceKatana450.xml	2012-06-29 10:43:01.711570452 +0200
@@ -126,7 +126,7 @@
         <pose v1="0.0" v2="0.09415" v3="0.0" roll="0.0" pitch="0.0" yaw="0.0"/>
       </bounds>
     </joint>
-    <gripper name="Gripper" maxCommTime="0.015">
+    <gripper name="Gripper" maxCommTime="0.115">
       <graspRange encoder_open="99000" encoder_closed="84200"/>
       <KNI angleOffset="-119.5" angleRange="329.0" encoderOffset="100000" encodersPerCycle="51200" rotationDirection="+1" encoderPositionAfter="99000" offset="1.570796327" gain="1.0"></KNI>
       <AxNI>
