diff -ru gazebo-0.9.0-orig/server/physics/Body.cc gazebo-0.9.0/server/physics/Body.cc
--- gazebo-0.9.0-orig/server/physics/Body.cc	2009-07-08 20:00:39.000000000 +0200
+++ gazebo-0.9.0/server/physics/Body.cc	2011-07-14 10:29:14.486227821 +0200
@@ -230,7 +230,7 @@
   if (this->mass.mass > 0.0)
   {
     std::ostringstream visname;
-    visname << this->GetName() << "_CGVISUAL" ;
+    visname << this->GetScopedName() + ":" + this->GetName() << "_CGVISUAL" ;
 
     this->cgVisual = OgreCreator::Instance()->CreateVisual(visname.str(),
                         this->GetModel()->GetVisualNode());
